subworkflows/local/prepare_genome.nf

```diff
@@ -27,110 +27,73 @@ workflow PREPARE_GENOME {
         germline_resource // channel: [optional]  germline_resource
         known_indels      // channel: [optional]  known_indels
         pon               // channel: [optional]  pon
-        tools             // value:   [mandatory] tools
-        step              // value:   [mandatory] step
 
     main:
 
-    if(!tools) tools = ""
-
     ch_versions = Channel.empty()
 
-    ch_bwa = Channel.empty()
-    if (!(params.bwa) && 'mapping' in step) {
-        if (params.aligner == "bwa-mem") {
-            BWAMEM1_INDEX(fasta)
-            ch_bwa = BWAMEM1_INDEX.out.index
-            ch_versions = ch_versions.mix(BWAMEM1_INDEX.out.versions)
-        } else if (params.aligner == "bwa-mem2") {
-            BWAMEM2_INDEX(fasta)
-            ch_bwa = BWAMEM2_INDEX.out.index
-            ch_versions = ch_versions.mix(BWAMEM2_INDEX.out.versions)
-        }
-    }
+    (ch_bwa1, ch_bwa_version) = BWAMEM1_INDEX(fasta)
+    (ch_bwa2, ch_bwa_version) = BWAMEM2_INDEX(fasta)
 
-    ch_dict = Channel.empty()
-    if (!(params.dict) && !('annotate' in step) && !('controlfreec' in step)) {
-        GATK4_CREATESEQUENCEDICTIONARY(fasta)
-        ch_dict = GATK4_CREATESEQUENCEDICTIONARY.out.dict
-        ch_versions = ch_versions.mix(GATK4_CREATESEQUENCEDICTIONARY.out.versions)
-    }
+    ch_bwa = ch_bwa1.mix(ch_bwa2)
+    ch_versions = ch_versions.mix(ch_bwa_version)
 
-    ch_fasta_fai = Channel.empty()
-    if (fasta_fai) ch_fasta_fai = fasta_fai
-    if (!(params.fasta_fai) && !('annotate' in step)) {
-        SAMTOOLS_FAIDX(fasta.map{ it -> [[id:it[0].getName()], it]})
-        ch_fasta_fai = SAMTOOLS_FAIDX.out.fai.map{ meta, fai -> [fai] }
-        ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)
-    }
+    (ch_dict, ch_dict_version) = GATK4_CREATESEQUENCEDICTIONARY(fasta)
 
-    ch_dbsnp_tbi = Channel.empty()
-    if (!(params.dbsnp_tbi) && params.dbsnp && ('mapping' in step || 'prepare_recalibration' in step || tools.contains('controlfreec') || tools.contains('haplotypecaller') || tools.contains('mutect2') || tools.contains('tnscope'))) {
-        TABIX_DBSNP(dbsnp.map{ it -> [[id:it[0].baseName], it] })
-        ch_dbsnp_tbi = TABIX_DBSNP.out.tbi.map{ meta, tbi -> [tbi] }
-        ch_versions = ch_versions.mix(TABIX_DBSNP.out.versions)
-    }
+    ch_versions = ch_versions.mix(ch_dict_version)
 
-    ch_germline_resource_tbi = Channel.empty()
-    if (!(params.germline_resource_tbi) && params.germline_resource && tools.contains('mutect2')) {
-        TABIX_GERMLINE_RESOURCE(germline_resource.map{ it -> [[id:it[0].baseName], it] })
-        ch_germline_resource_tbi = TABIX_GERMLINE_RESOURCE.out.tbi.map{ meta, tbi -> [tbi] }
-        ch_versions = ch_versions.mix(TABIX_GERMLINE_RESOURCE.out.versions)
+    if (fasta_fai) ch_fasta_fai = fasta_fai
```

Comment:
I see now that we are somehow fairly inconsistent if we pass things over as input or not. Maybe we can unify this